src/org/zalando/stups/mint/worker/external/bucket_storage.clj

```diff
@@ -0,0 +1,26 @@
+(ns org.zalando.stups.mint.worker.external.bucket_storage)
+
+(defmacro StorageException
+  [msg data]
+  `(ex-info ~msg (merge ~data {:type "StorageException"})))
+
+(defn storage-exception? [e]
+  (= "StorageException" (:type (ex-data e))))
+
+; TODO: actually infer the bucket type
+(defn infer-bucket-type
+  "infer bucket type from bucket name"
+  [bucket-name]
+  (if (.startsWith bucket-name "gs://") :gs :s3))
+
+(defmulti writable?
+          "Check if a bucket is writable?"
+          (fn [x] (infer-bucket-type (x :bucket-name))))
```

Comment:
the `fn` can also have multiple parameters, e.g. I would like to keep the signature: `(writable? bucket-name app-id)`. So the multi method could look like:

```clojure
(defmulti writable?
          "Check if a bucket is writable?"
          (fn [bucket-name _] (infer-bucket-type bucket-name))

(defmethod writable? :s3 [bucket-name app-id]
    ...)

(defmethod writable? :gs [bucket-name app-id]
    ...)
```